src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                zero_boundaries=False,
+            )
+
+        elif "GaussianSquare" in str(pulse.shape):
+            sigma = pulse.shape.rel_sigma
+            zh_pulse = lo.pulse_library.gaussian_square(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                width=round(pulse.duration * 1e-9, 9) * 0.9,  # 90% Flat
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                zero_boundaries=False,
+            )
+
+        elif "Drag" in str(pulse.shape):
+            sigma = pulse.shape.rel_sigma
+            beta = pulse.shape.beta
+            zh_pulse = lo.pulse_library.drag(
+                uid=(f"{pulse_type}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+                sigma=2 / sigma,
+                beta=beta,
+                zero_boundaries=False,
+            )
+        elif "Slepian" in str(pulse.shape):
+            "Implement Slepian shaped flux pulse https://arxiv.org/pdf/0909.5368.pdf"
+
+        elif "Sampled" in str(pulse.shape):
+            "Implement Sampled pulses for Optimal control algorithms like GRAPE"
+
+            """
+            Typically, the sampler function should discard ``length`` and ``amplitude``, and
+            instead assume that the pulse extends from -1 to 1, and that it has unit
+            amplitude. LabOne Q will automatically rescale the sampler's output to the correct
+            amplitude and length.
+
+            They don't even do that on their notebooks
+            and just use lenght and amplitude but we have to check
+
+            x = pulse.envelope_waveform_i.data  No need for q ???
+            """
+
+        return zh_pulse
+
+
+class ZhSweeper:
+    """
+    Zurich sweeper from qibolab sweeper for pulse parameters
+    Amplitude, Duration, Frequency (and maybe Phase)
+
+    """
+
+    def __init__(self, pulse, sweeper, qubit):
+        """Qibolab sweeper"""
+        self.sweeper = sweeper
+
+        """Qibolab pulse associated to the sweeper"""
+        self.pulse = pulse
+        """Line associated with the pulse"""
+        self.signal = f"{pulse.type.name.lower()}{pulse.qubit}"
+        """Zurich pulse associated to the sweeper"""
+        self.zhpulse = ZhPulse(pulse).zhpulse
+
+        """Zurich sweeper"""
+        self.zhsweeper = self.select_sweeper(sweeper, qubit)
+
+        """
+        Zurich sweepers, Need something better to store multiple sweeps on the same pulse
+
+        Not properly implemented as it was only used on Rabi amplitude vs lenght and it
+        was an unused routine.
+        """
+        self.zhsweepers = [self.select_sweeper(sweeper, qubit)]
+
+    def select_sweeper(self, sweeper, qubit):
```

Comment:
This function is only using `self.pulse.type` attribute of `self`, and nothing else.

So, maybe you could move out of the class:
```suggestion
    def select_sweeper(ptype, sweeper, qubit):
```

Reply:
Actually, make it a static method, as written below:
```suggestion
    @staticmethod
    def select_sweeper(ptype, sweeper, qubit):
```